apps/recovery-relay/lib/wallets/XRP/index.ts

```diff
@@ -63,9 +63,16 @@ export class Ripple extends BaseRipple implements ConnectedWallet {
       await this.xrpClient.connect();
     }
     try {
-      const txRes = (await this.xrpClient.submitAndWait(tx)) as TxResponse;
-      this.relayLogger.debug(`Ripple: Tx broadcasted: ${JSON.stringify(txRes, null, 2)}`);
-      return txRes.result.hash;
+      const { result: txResult } = (await this.xrpClient.submit(tx)) as SubmitResponse;
+      this.relayLogger.debug(`Ripple: Tx broadcasted: ${JSON.stringify(txResult, null, 2)}`);
+      const { hash } = txResult.tx_json;
+      if (!hash) {
+        if (!txResult.accepted) {
```

Comment:
I'm just leaving comments to nitpick because otherwise everything else looks fine. I just like this syntax. You don't need to change if you don't want to. 

    if (!hash) {
        const errorMessage = txResult.accepted 
            ? "Transaction didn't yield hash"
            : `Transaction not accepted: ${txResult.engine_result_message}`;
        throw new Error(errorMessage);
    }

Reply:
Updated